integration-tests/fault-injection/harness/inserter.go

```diff
@@ -21,17 +21,23 @@ func RunInserts(client *mongo.Database, numInserts int, frequency time.Duration)
 	for i := 0; i < numInserts; i++ {
 		id := fmt.Sprintf("doc%d", i)
 
-		// We set a 50ms timout for the insert: long enough that the insert will
+		// We set a 100ms timeout for the insert: long enough that the insert will
 		// succeed if Mongo is working normally, but too short for it to retry during
-		// a failover
-		ctx, cancel := context.WithTimeout(context.Background(), 50*time.Millisecond)
+		// a failover.
+
+		// The write may still get through even if the InsertOne call errors out and if the resulting InsertedID is nil.
+		ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
 		defer cancel()
-		_, err := client.Collection("Test").InsertOne(ctx, bson.M{
+		insertResult, err := client.Collection("Test").InsertOne(ctx, bson.M{
 			"_id": id,
 		})
 
 		if err != nil {
-			log.Printf("Warning: mongo insert failed: %s", err)
+			log.Printf("Warning: mongo insert failed for doc %s: %s", id, err)
+			if insertResult != nil && insertResult.InsertedID != nil {
+				log.Printf("Warning: although the previous insert faced this error, the InsertedID wasn't nil, so we'll conclude it was a success (InterruptedDueToReplStateChange). InsertedID: %s", insertResult.InsertedID)
```

Comment:
🤔 is there a way to check the type of `err`  (e.g., for `InterruptedDueToReplStateChange`)?

Reply:
I guess, this worked though, so lets go with this one. There could be many different errors, this was just what I picked up. The result is apparently simply not reliable in case of insert when using replicated mongo.